// @pipecat-ai/client-js/dist/index.js
var $58KWk$events = require("events");
var $58KWk$uuid = require("uuid");
var $58KWk$bowser = require("bowser");


function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || Object.prototype.hasOwnProperty.call(dest, key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ var $c2758f3146a53d04$exports = {};
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ var $02f900e1428a591c$exports = {};

$parcel$export($02f900e1428a591c$exports, "PipecatClient", () => $02f900e1428a591c$export$8f7f86a77535f7a3);
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ 
var $8cf63be2a2a16953$exports = {};
$8cf63be2a2a16953$exports = JSON.parse("{\"name\":\"@pipecat-ai/client-js\",\"version\":\"1.7.0\",\"license\":\"BSD-2-Clause\",\"main\":\"dist/index.js\",\"module\":\"dist/index.module.js\",\"types\":\"dist/index.d.ts\",\"source\":\"index.ts\",\"repository\":{\"type\":\"git\",\"url\":\"git+https://github.com/pipecat-ai/pipecat-client-web.git\"},\"exports\":{\".\":{\"types\":\"./dist/index.d.ts\",\"import\":\"./dist/index.module.js\",\"require\":\"./dist/index.js\"}},\"files\":[\"dist\",\"package.json\",\"README.md\"],\"scripts\":{\"build\":\"jest --silent --passWithNoTests && parcel build --no-cache\",\"dev\":\"parcel watch\",\"lint\":\"eslint . --report-unused-disable-directives --max-warnings 0\",\"test\":\"jest\"},\"jest\":{\"preset\":\"ts-jest\",\"testEnvironment\":\"jsdom\",\"setupFilesAfterEnv\":[\"<rootDir>/tests/jest.setup.ts\"]},\"devDependencies\":{\"@jest/globals\":\"^29.7.0\",\"@types/clone-deep\":\"^4.0.4\",\"@types/jest\":\"^29.5.12\",\"@types/uuid\":\"^10.0.0\",\"eslint\":\"^9.11.1\",\"eslint-config-prettier\":\"^9.1.0\",\"eslint-plugin-simple-import-sort\":\"^12.1.1\",\"jest\":\"^29.7.0\",\"jest-environment-jsdom\":\"^30.0.2\",\"ts-jest\":\"^29.2.5\",\"whatwg-fetch\":\"^3.6.20\"},\"dependencies\":{\"@types/events\":\"^3.0.3\",\"bowser\":\"^2.11.0\",\"clone-deep\":\"^4.0.1\",\"events\":\"^3.3.0\",\"typed-emitter\":\"^2.1.0\",\"uuid\":\"^10.0.0\"}}");


var $609db2a766ef81cc$exports = {};
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ var $920632f0518a5f1b$exports = {};

$parcel$export($920632f0518a5f1b$exports, "TransportStateEnum", () => $920632f0518a5f1b$export$8f2038d3679a1d9b);
var $920632f0518a5f1b$export$8f2038d3679a1d9b;
(function(TransportStateEnum) {
    TransportStateEnum["DISCONNECTED"] = "disconnected";
    TransportStateEnum["INITIALIZING"] = "initializing";
    TransportStateEnum["INITIALIZED"] = "initialized";
    TransportStateEnum["AUTHENTICATING"] = "authenticating";
    TransportStateEnum["AUTHENTICATED"] = "authenticated";
    TransportStateEnum["CONNECTING"] = "connecting";
    TransportStateEnum["CONNECTED"] = "connected";
    TransportStateEnum["READY"] = "ready";
    TransportStateEnum["DISCONNECTING"] = "disconnecting";
    TransportStateEnum["ERROR"] = "error";
})($920632f0518a5f1b$export$8f2038d3679a1d9b || ($920632f0518a5f1b$export$8f2038d3679a1d9b = {}));


var $b4e51e049033e221$exports = {};

$parcel$export($b4e51e049033e221$exports, "RTVIError", () => $b4e51e049033e221$export$59b4786f333aac02);
$parcel$export($b4e51e049033e221$exports, "ConnectionTimeoutError", () => $b4e51e049033e221$export$c67992fa684a81a6);
$parcel$export($b4e51e049033e221$exports, "StartBotError", () => $b4e51e049033e221$export$e7544ab812238a61);
$parcel$export($b4e51e049033e221$exports, "TransportStartError", () => $b4e51e049033e221$export$e0624a511a2c4e9);
$parcel$export($b4e51e049033e221$exports, "InvalidTransportParamsError", () => $b4e51e049033e221$export$b6ce555ea7f95fba);
$parcel$export($b4e51e049033e221$exports, "BotNotReadyError", () => $b4e51e049033e221$export$885fb96b850e8fbb);
$parcel$export($b4e51e049033e221$exports, "BotAlreadyStartedError", () => $b4e51e049033e221$export$cc240eab14fa4f50);
$parcel$export($b4e51e049033e221$exports, "UnsupportedFeatureError", () => $b4e51e049033e221$export$bd0820eb8444fcd9);
$parcel$export($b4e51e049033e221$exports, "MessageTooLargeError", () => $b4e51e049033e221$export$78e1011ee1942cf6);
$parcel$export($b4e51e049033e221$exports, "DeviceError", () => $b4e51e049033e221$export$64c9f614187c1e59);
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ class $b4e51e049033e221$export$59b4786f333aac02 extends Error {
    constructor(message, status){
        super(message);
        this.status = status;
    }
}
class $b4e51e049033e221$export$c67992fa684a81a6 extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(message){
        super(message ?? "Bot did not enter ready state within the specified timeout period.");
    }
}
class $b4e51e049033e221$export$e7544ab812238a61 extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(message, status){
        super(message ?? `Failed to connect / invalid auth bundle from base url`, status ?? 500);
        this.error = "invalid-request-error";
    }
}
class $b4e51e049033e221$export$e0624a511a2c4e9 extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(message){
        super(message ?? "Unable to connect to transport");
    }
}
class $b4e51e049033e221$export$b6ce555ea7f95fba extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(message){
        super(message ?? "Invalid transport connection parameters");
    }
}
class $b4e51e049033e221$export$885fb96b850e8fbb extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(message){
        super(message ?? "Attempt to call action on transport when not in 'ready' state.");
    }
}
class $b4e51e049033e221$export$cc240eab14fa4f50 extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(message){
        super(message ?? "Pipecat client has already been started. Please call disconnect() before starting again.");
    }
}
class $b4e51e049033e221$export$bd0820eb8444fcd9 extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(feature, source, message){
        let msg = `${feature} not supported${message ? `: ${message}` : ""}`;
        if (source) msg = `${source} does not support ${feature}${message ? `: ${message}` : ""}`;
        super(msg);
        this.feature = feature;
    }
}
class $b4e51e049033e221$export$78e1011ee1942cf6 extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(message){
        super(message ?? "Message size exceeds the maximum allowed limit for transport.");
    }
}
class $b4e51e049033e221$export$64c9f614187c1e59 extends $b4e51e049033e221$export$59b4786f333aac02 {
    constructor(devices, type, message, details){
        super(message ?? `Device error for ${devices.join(", ")}: ${type}`);
        this.devices = devices;
        this.type = type;
        this.details = details;
    }
}


var $82c0e475b1b90003$exports = {};

$parcel$export($82c0e475b1b90003$exports, "RTVIEvent", () => $82c0e475b1b90003$export$6b4624d233c61fcb);
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ var $82c0e475b1b90003$export$6b4624d233c61fcb;
(function(RTVIEvent) {
    /** local connection state events */ RTVIEvent["Connected"] = "connected";
    RTVIEvent["Disconnected"] = "disconnected";
    RTVIEvent["TransportStateChanged"] = "transportStateChanged";
    /** remote connection state events */ RTVIEvent["BotStarted"] = "botStarted";
    RTVIEvent["BotConnected"] = "botConnected";
    RTVIEvent["BotReady"] = "botReady";
    RTVIEvent["BotDisconnected"] = "botDisconnected";
    RTVIEvent["Error"] = "error";
    /** server messaging */ RTVIEvent["ServerMessage"] = "serverMessage";
    RTVIEvent["ServerResponse"] = "serverResponse";
    RTVIEvent["MessageError"] = "messageError";
    /** service events */ RTVIEvent["Metrics"] = "metrics";
    // vad events
    RTVIEvent["BotStartedSpeaking"] = "botStartedSpeaking";
    RTVIEvent["BotStoppedSpeaking"] = "botStoppedSpeaking";
    RTVIEvent["UserStartedSpeaking"] = "userStartedSpeaking";
    RTVIEvent["UserStoppedSpeaking"] = "userStoppedSpeaking";
    // user mute strategy events
    RTVIEvent["UserMuteStarted"] = "userMuteStarted";
    RTVIEvent["UserMuteStopped"] = "userMuteStopped";
    // stt events
    RTVIEvent["UserTranscript"] = "userTranscript";
    RTVIEvent["BotOutput"] = "botOutput";
    // DEPRECATED
    RTVIEvent["BotTranscript"] = "botTranscript";
    // llm events
    RTVIEvent["BotLlmText"] = "botLlmText";
    RTVIEvent["BotLlmStarted"] = "botLlmStarted";
    RTVIEvent["BotLlmStopped"] = "botLlmStopped";
    // DEPRECATED
    RTVIEvent["LLMFunctionCall"] = "llmFunctionCall";
    RTVIEvent["LLMFunctionCallStarted"] = "llmFunctionCallStarted";
    RTVIEvent["LLMFunctionCallInProgress"] = "llmFunctionCallInProgress";
    RTVIEvent["LLMFunctionCallStopped"] = "llmFunctionCallStopped";
    RTVIEvent["BotLlmSearchResponse"] = "botLlmSearchResponse";
    // tts events
    RTVIEvent["BotTtsText"] = "botTtsText";
    RTVIEvent["BotTtsStarted"] = "botTtsStarted";
    RTVIEvent["BotTtsStopped"] = "botTtsStopped";
    /** participant events */ RTVIEvent["ParticipantConnected"] = "participantConnected";
    RTVIEvent["ParticipantLeft"] = "participantLeft";
    /** media events */ RTVIEvent["TrackStarted"] = "trackStarted";
    RTVIEvent["TrackStopped"] = "trackStopped";
    RTVIEvent["ScreenTrackStarted"] = "screenTrackStarted";
    RTVIEvent["ScreenTrackStopped"] = "screenTrackStopped";
    RTVIEvent["ScreenShareError"] = "screenShareError";
    RTVIEvent["LocalAudioLevel"] = "localAudioLevel";
    RTVIEvent["RemoteAudioLevel"] = "remoteAudioLevel";
    /** media device events */ RTVIEvent["AvailableCamsUpdated"] = "availableCamsUpdated";
    RTVIEvent["AvailableMicsUpdated"] = "availableMicsUpdated";
    RTVIEvent["AvailableSpeakersUpdated"] = "availableSpeakersUpdated";
    RTVIEvent["CamUpdated"] = "camUpdated";
    RTVIEvent["MicUpdated"] = "micUpdated";
    RTVIEvent["SpeakerUpdated"] = "speakerUpdated";
    RTVIEvent["DeviceError"] = "deviceError";
})($82c0e475b1b90003$export$6b4624d233c61fcb || ($82c0e475b1b90003$export$6b4624d233c61fcb = {}));


var $b1a2fe36be912934$exports = {};

$parcel$export($b1a2fe36be912934$exports, "RTVI_PROTOCOL_VERSION", () => $b1a2fe36be912934$export$7bdaf0e0d661a8f5);
$parcel$export($b1a2fe36be912934$exports, "RTVI_MESSAGE_LABEL", () => $b1a2fe36be912934$export$882b13c7fda338f5);
$parcel$export($b1a2fe36be912934$exports, "RTVIMessageType", () => $b1a2fe36be912934$export$38b3db05cbf0e240);
$parcel$export($b1a2fe36be912934$exports, "AggregationType", () => $b1a2fe36be912934$export$fa4739a8a27f18c0);
$parcel$export($b1a2fe36be912934$exports, "setAboutClient", () => $b1a2fe36be912934$export$e4036f9b8ddb7379);
$parcel$export($b1a2fe36be912934$exports, "RTVIMessage", () => $b1a2fe36be912934$export$69aa9ab0334b212);
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ 

const $b1a2fe36be912934$export$7bdaf0e0d661a8f5 = "1.2.0";
const $b1a2fe36be912934$export$882b13c7fda338f5 = "rtvi-ai";
var $b1a2fe36be912934$export$38b3db05cbf0e240;
(function(RTVIMessageType) {
    /** Outbound Messages */ RTVIMessageType["CLIENT_READY"] = "client-ready";
    RTVIMessageType["DISCONNECT_BOT"] = "disconnect-bot";
    // Client-to-server messages
    RTVIMessageType["CLIENT_MESSAGE"] = "client-message";
    RTVIMessageType["SEND_TEXT"] = "send-text";
    // DEPRECATED
    RTVIMessageType["APPEND_TO_CONTEXT"] = "append-to-context";
    /**
     * Inbound Messages
     * Messages the server-side client sends to our client-side client regarding
     * its state or other non-service-specific messaging.
     */ RTVIMessageType["BOT_READY"] = "bot-ready";
    RTVIMessageType["ERROR"] = "error";
    RTVIMessageType["METRICS"] = "metrics";
    RTVIMessageType["SERVER_MESSAGE"] = "server-message";
    RTVIMessageType["SERVER_RESPONSE"] = "server-response";
    RTVIMessageType["ERROR_RESPONSE"] = "error-response";
    RTVIMessageType["APPEND_TO_CONTEXT_RESULT"] = "append-to-context-result";
    /** Speaking and Transcription Messages */ RTVIMessageType["USER_STARTED_SPEAKING"] = "user-started-speaking";
    RTVIMessageType["USER_STOPPED_SPEAKING"] = "user-stopped-speaking";
    RTVIMessageType["BOT_STARTED_SPEAKING"] = "bot-started-speaking";
    RTVIMessageType["BOT_STOPPED_SPEAKING"] = "bot-stopped-speaking";
    // User muted events. These events notify when the server is ignoring audio from the client.
    // The client should continue sending audio normally but may want to show some indication to the user.
    RTVIMessageType["USER_MUTE_STARTED"] = "user-mute-started";
    RTVIMessageType["USER_MUTE_STOPPED"] = "user-mute-stopped";
    RTVIMessageType["USER_TRANSCRIPTION"] = "user-transcription";
    RTVIMessageType["BOT_OUTPUT"] = "bot-output";
    // DEPRECATED
    RTVIMessageType["BOT_TRANSCRIPTION"] = "bot-transcription";
    /** LLM Messages */ RTVIMessageType["USER_LLM_TEXT"] = "user-llm-text";
    RTVIMessageType["BOT_LLM_TEXT"] = "bot-llm-text";
    RTVIMessageType["BOT_LLM_STARTED"] = "bot-llm-started";
    RTVIMessageType["BOT_LLM_STOPPED"] = "bot-llm-stopped";
    // Function calling
    // DECPRECATED
    RTVIMessageType["LLM_FUNCTION_CALL"] = "llm-function-call";
    RTVIMessageType["LLM_FUNCTION_CALL_STARTED"] = "llm-function-call-started";
    RTVIMessageType["LLM_FUNCTION_CALL_IN_PROGRESS"] = "llm-function-call-in-progress";
    RTVIMessageType["LLM_FUNCTION_CALL_STOPPED"] = "llm-function-call-stopped";
    RTVIMessageType["LLM_FUNCTION_CALL_RESULT"] = "llm-function-call-result";
    RTVIMessageType["BOT_LLM_SEARCH_RESPONSE"] = "bot-llm-search-response";
    /** TTS Messages */ RTVIMessageType["BOT_TTS_TEXT"] = "bot-tts-text";
    RTVIMessageType["BOT_TTS_STARTED"] = "bot-tts-started";
    RTVIMessageType["BOT_TTS_STOPPED"] = "bot-tts-stopped";
})($b1a2fe36be912934$export$38b3db05cbf0e240 || ($b1a2fe36be912934$export$38b3db05cbf0e240 = {}));
var $b1a2fe36be912934$export$fa4739a8a27f18c0;
(function(AggregationType) {
    AggregationType["WORD"] = "word";
    AggregationType["SENTENCE"] = "sentence";
})($b1a2fe36be912934$export$fa4739a8a27f18c0 || ($b1a2fe36be912934$export$fa4739a8a27f18c0 = {}));
// ----- Message Classes
let $b1a2fe36be912934$var$_aboutClient;
function $b1a2fe36be912934$export$e4036f9b8ddb7379(about) {
    // allow for partial updates to the about data
    // this allows the client to set the about data at any time
    // before sending the `client-ready` message and not worry about
    // overwriting existing data
    if ($b1a2fe36be912934$var$_aboutClient) $b1a2fe36be912934$var$_aboutClient = {
        ...$b1a2fe36be912934$var$_aboutClient,
        ...about
    };
    else // if no about data is set, set it to the provided value
    $b1a2fe36be912934$var$_aboutClient = about;
}
class $b1a2fe36be912934$export$69aa9ab0334b212 {
    constructor(type, data, id){
        this.label = $b1a2fe36be912934$export$882b13c7fda338f5;
        this.type = type;
        this.data = data;
        this.id = id || (0, $58KWk$uuid.v4)().slice(0, 8);
    }
    // Outbound message types
    static clientReady() {
        return new $b1a2fe36be912934$export$69aa9ab0334b212($b1a2fe36be912934$export$38b3db05cbf0e240.CLIENT_READY, {
            version: $b1a2fe36be912934$export$7bdaf0e0d661a8f5,
            about: $b1a2fe36be912934$var$_aboutClient || {
                library: (0, $8cf63be2a2a16953$exports.name),
                library_version: (0, $8cf63be2a2a16953$exports.version)
            }
        });
    }
    static disconnectBot() {
        return new $b1a2fe36be912934$export$69aa9ab0334b212($b1a2fe36be912934$export$38b3db05cbf0e240.DISCONNECT_BOT, {});
    }
    static error(message, fatal = false) {
        return new $b1a2fe36be912934$export$69aa9ab0334b212($b1a2fe36be912934$export$38b3db05cbf0e240.ERROR, {
            message: message,
            fatal: fatal
        });
    }
}


$parcel$exportWildcard($609db2a766ef81cc$exports, $920632f0518a5f1b$exports);
$parcel$exportWildcard($609db2a766ef81cc$exports, $b4e51e049033e221$exports);
$parcel$exportWildcard($609db2a766ef81cc$exports, $82c0e475b1b90003$exports);
$parcel$exportWildcard($609db2a766ef81cc$exports, $b1a2fe36be912934$exports);



/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ 
function $37a63b97d182bc86$export$f1586721024c4dab(_target, propertyKey, descriptor) {
    const originalMethod = descriptor.value;
    descriptor.value = function(...args) {
        if (this.state === "ready") return originalMethod.apply(this, args);
        else throw new (0, $b4e51e049033e221$export$885fb96b850e8fbb)(`Attempt to call ${propertyKey.toString()} when transport not in ready state. Await connect() first.`);
    };
    return descriptor;
}
function $37a63b97d182bc86$export$ebc0d747cf8770bc(_target, propertyKey, descriptor) {
    const originalMethod = descriptor.value;
    const states = [
        "authenticating",
        "connecting",
        "connected",
        "ready"
    ];
    descriptor.value = function(...args) {
        if (states.includes(this.state)) throw new (0, $b4e51e049033e221$export$cc240eab14fa4f50)(`Attempt to call ${propertyKey.toString()} when client already started. Please call disconnect() before starting again.`);
        else return originalMethod.apply(this, args);
    };
    return descriptor;
}
function $37a63b97d182bc86$export$808994d0d8c9acb3(states) {
    return function(_target, propertyKey, descriptor) {
        const originalMethod = descriptor.value;
        descriptor.get = function(...args) {
            if (states.includes(this.state)) return originalMethod.apply(this, args);
            else throw new (0, $b4e51e049033e221$export$885fb96b850e8fbb)(`Attempt to call ${propertyKey.toString()} when transport not in ${states}.`);
        };
        return descriptor;
    };
}
function $37a63b97d182bc86$export$5c35b4fe6fa8c9a6(...states) {
    states = [
        "ready",
        ...states
    ];
    return function(_target, propertyKey, descriptor) {
        const originalGetter = descriptor.get;
        descriptor.get = function() {
            if (states.includes(this.state)) return originalGetter?.apply(this);
            else throw new (0, $b4e51e049033e221$export$885fb96b850e8fbb)(`Attempt to call ${propertyKey.toString()} when transport not in ${states}. Await connect() first.`);
        };
        return descriptor;
    };
}


var $3b981481778defd6$exports = {};

$parcel$export($3b981481778defd6$exports, "MessageDispatcher", () => $3b981481778defd6$export$e9a960646cc432aa);
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ 
var $42491a06a064f4c0$exports = {};

$parcel$export($42491a06a064f4c0$exports, "LogLevel", () => $42491a06a064f4c0$export$243e62d78d3b544d);
$parcel$export($42491a06a064f4c0$exports, "logger", () => $42491a06a064f4c0$export$af88d00dbe7f521);
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ var $42491a06a064f4c0$export$243e62d78d3b544d;
(function(LogLevel) {
    LogLevel[LogLevel["NONE"] = 0] = "NONE";
    LogLevel[LogLevel["ERROR"] = 1] = "ERROR";
    LogLevel[LogLevel["WARN"] = 2] = "WARN";
    LogLevel[LogLevel["INFO"] = 3] = "INFO";
    LogLevel[LogLevel["DEBUG"] = 4] = "DEBUG";
})($42491a06a064f4c0$export$243e62d78d3b544d || ($42491a06a064f4c0$export$243e62d78d3b544d = {}));
class $42491a06a064f4c0$var$Logger {
    constructor(){
        this.level = $42491a06a064f4c0$export$243e62d78d3b544d.DEBUG;
    }
    static getInstance() {
        if (!$42491a06a064f4c0$var$Logger.instance) $42491a06a064f4c0$var$Logger.instance = new $42491a06a064f4c0$var$Logger();
        return $42491a06a064f4c0$var$Logger.instance;
    }
    setLevel(level) {
        this.level = level;
    }
    debug(...args) {
        if (this.level >= $42491a06a064f4c0$export$243e62d78d3b544d.DEBUG) console.debug(...args);
    }
    info(...args) {
        if (this.level >= $42491a06a064f4c0$export$243e62d78d3b544d.INFO) console.info(...args);
    }
    warn(...args) {
        if (this.level >= $42491a06a064f4c0$export$243e62d78d3b544d.WARN) console.warn(...args);
    }
    error(...args) {
        if (this.level >= $42491a06a064f4c0$export$243e62d78d3b544d.ERROR) console.error(...args);
    }
}
const $42491a06a064f4c0$export$af88d00dbe7f521 = $42491a06a064f4c0$var$Logger.getInstance();


class $3b981481778defd6$export$e9a960646cc432aa {
    constructor(sendMethod){
        this._queue = new Array();
        this._gcInterval = undefined;
        this._queue = [];
        this._sendMethod = sendMethod;
    }
    disconnect() {
        this.clearQueue();
        clearInterval(this._gcInterval);
        this._gcInterval = undefined;
    }
    dispatch(message_data, type = (0, $b1a2fe36be912934$export$38b3db05cbf0e240).CLIENT_MESSAGE, timeout = 10000) {
        if (!this._gcInterval) // start garbage collection if not already running
        this._gcInterval = setInterval(()=>{
            this._gc();
        }, 2000); // Run garbage collection every 2 seconds
        const message = new (0, $b1a2fe36be912934$export$69aa9ab0334b212)(type, message_data);
        const promise = new Promise((resolve, reject)=>{
            this._queue.push({
                message: message,
                timestamp: Date.now(),
                timeout: timeout,
                resolve: resolve,
                reject: reject
            });
        });
        (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[MessageDispatcher] dispatch", message);
        try {
            this._sendMethod(message);
        } catch (e) {
            (0, $42491a06a064f4c0$export$af88d00dbe7f521).error("[MessageDispatcher] Error sending message", e);
            return Promise.reject(e);
        }
        this._gc();
        return promise;
    }
    clearQueue() {
        this._queue = [];
    }
    _resolveReject(message, resolve = true) {
        const queuedMessage = this._queue.find((msg)=>msg.message.id === message.id);
        if (queuedMessage) {
            if (resolve) {
                (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[MessageDispatcher] Resolve", message);
                queuedMessage.resolve(message);
            } else {
                (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[MessageDispatcher] Reject", message);
                queuedMessage.reject(message);
            }
            // Remove message from queue
            this._queue = this._queue.filter((msg)=>msg.message.id !== message.id);
            (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[MessageDispatcher] Queue", this._queue);
        }
        return message;
    }
    resolve(message) {
        return this._resolveReject(message, true);
    }
    reject(message) {
        return this._resolveReject(message, false);
    }
    _gc() {
        const expired = [];
        this._queue = this._queue.filter((msg)=>{
            const isValid = Date.now() - msg.timestamp < msg.timeout;
            if (!isValid) expired.push(msg);
            return isValid;
        });
        expired.forEach((msg)=>{
            if (msg.message.type === (0, $b1a2fe36be912934$export$38b3db05cbf0e240).CLIENT_MESSAGE) msg.reject(new (0, $b1a2fe36be912934$export$69aa9ab0334b212)((0, $b1a2fe36be912934$export$38b3db05cbf0e240).ERROR_RESPONSE, {
                error: "Timed out waiting for response",
                msgType: msg.message.data.t,
                data: msg.message.data.d,
                fatal: false
            }));
        });
        (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[MessageDispatcher] GC", this._queue);
    }
}



var $e93e1075413e9e2e$exports = {};

$parcel$export($e93e1075413e9e2e$exports, "isAPIRequest", () => $e93e1075413e9e2e$export$2dd7ca293b2783);
$parcel$export($e93e1075413e9e2e$exports, "makeRequest", () => $e93e1075413e9e2e$export$699251e5611cc6db);

function $e93e1075413e9e2e$export$2dd7ca293b2783(value) {
    if (typeof value === "object" && value !== null && Object.keys(value).includes("endpoint")) {
        const endpoint = value["endpoint"];
        return typeof endpoint === "string" || endpoint instanceof URL || typeof Request !== "undefined" && endpoint instanceof Request;
    }
    return false;
}
async function $e93e1075413e9e2e$export$699251e5611cc6db(cxnOpts, abortController) {
    if (!abortController) abortController = new AbortController();
    let handshakeTimeout;
    return new Promise((resolve, reject)=>{
        (async ()=>{
            if (cxnOpts.timeout) handshakeTimeout = setTimeout(async ()=>{
                abortController.abort();
                reject(new Error("Timed out"));
            }, cxnOpts.timeout);
            let request;
            if (typeof Request !== "undefined" && cxnOpts.endpoint instanceof Request) {
                request = new Request(cxnOpts.endpoint, {
                    signal: abortController.signal
                });
                if (cxnOpts.requestData) (0, $42491a06a064f4c0$export$af88d00dbe7f521).warn("[Pipecat Client] requestData in APIRequest is ignored when endpoint is a Request object");
                if (cxnOpts.headers) (0, $42491a06a064f4c0$export$af88d00dbe7f521).warn("[Pipecat Client] headers in APIRequest is ignored when endpoint is a Request object");
            } else request = new Request(cxnOpts.endpoint, {
                method: "POST",
                mode: "cors",
                headers: new Headers({
                    "Content-Type": "application/json",
                    ...Object.fromEntries((cxnOpts.headers ?? new Headers()).entries())
                }),
                body: JSON.stringify(cxnOpts.requestData),
                signal: abortController.signal
            });
            (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug(`[Pipecat Client] Fetching from ${request.url}`);
            fetch(request).then((res)=>{
                (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug(`[Pipecat Client] Received response from ${request.url}`, res);
                if (!res.ok) {
                    reject(res);
                    return;
                }
                return res.json();
            }).then((data)=>{
                resolve(data);
            }).catch((err)=>{
                (0, $42491a06a064f4c0$export$af88d00dbe7f521).error(`[Pipecat Client] Error fetching: ${err}`);
                reject(err);
            }).finally(()=>{
                if (handshakeTimeout) clearTimeout(handshakeTimeout);
            });
        })();
    });
}


var $8e196b651aab6862$exports = {};

$parcel$export($8e196b651aab6862$exports, "Transport", () => $8e196b651aab6862$export$86495b081fef8e52);
$parcel$export($8e196b651aab6862$exports, "TransportWrapper", () => $8e196b651aab6862$export$82b6ede160a64a3c);
/**
 * Copyright (c) 2024, Daily.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */ 
class $8e196b651aab6862$export$86495b081fef8e52 {
    constructor(){
        this._state = "disconnected";
        /**
         * Maximum allowed size in bytes for a single signaling/message payload.
         *
         * The default is 64 KiB (`64 * 1024`), which is chosen to stay well within
         * typical WebSocket, proxy, and intermediary limits and to discourage
         * transports from sending very large payloads in a single message.
         *
         * Transport implementations may override this value in subclasses or
         * constructors if their underlying transport has stricter or more relaxed
         * limits, as long as they continue to honor this field when enforcing
         * message size constraints.
         */ this._maxMessageSize = 65536; // 64 KiB
    }
    /**
     * Establishes a connection with the remote server. This is the main entry
     * point for the transport to start sending and receiving media and messages.
     * This is called from PipecatClient.connect() and should not be called directly.
     * @param connectParams - This type will ultimately be defned by the transport
     * implementation. It is used to pass connection parameters to the transport.
     */ connect(connectParams) {
        this._abortController = new AbortController();
        let validatedParams = connectParams;
        try {
            validatedParams = this._validateConnectionParams(connectParams);
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        } catch (e) {
            throw new (0, $b4e51e049033e221$export$59b4786f333aac02)(`Invalid connection params: ${e.message}. Please check your connection params and try again.`);
        }
        return this._connect(validatedParams);
    }
    /**
     * Allow the transports to determine how the bot was started.
     */ get startBotParams() {
        return this._startBotParams;
    }
    /**
     * Set the parameters used to start the bot.
     * @param startBotParams
     */ set startBotParams(startBotParams) {
        if (typeof Request !== "undefined" && startBotParams.endpoint instanceof Request) {
            // If the endpoint is a Request object, we have to clone it.
            // Otherwise, we might run into issues with body being already used.
            this._startBotParams = {
                ...startBotParams,
                endpoint: startBotParams.endpoint.clone()
            };
            return;
        }
        this._startBotParams = startBotParams;
    }
    /**
     * Disconnects the transport from the remote server. This is called from
     * PipecatClient.disconnect() and should not be called directly.
     */ disconnect() {
        if (this._abortController) this._abortController.abort();
        return this._disconnect();
    }
    /**
     * Maximum size, in bytes, of a single message that this transport will attempt
     * to send. Callers should ensure that any outbound {@link RTVIMessage} payloads
     * do not exceed this limit to avoid transport or server errors.
     */ get maxMessageSize() {
        return this._maxMessageSize;
    }
}
class $8e196b651aab6862$export$82b6ede160a64a3c {
    constructor(transport){
        this._transport = transport;
        this._proxy = new Proxy(this._transport, {
            get: (target, prop, receiver)=>{
                if (typeof target[prop] === "function") {
                    let errMsg;
                    switch(String(prop)){
                        // Disable methods that modify the lifecycle of the call. These operations
                        // should be performed via the Pipecat client in order to keep state in sync.
                        case "initialize":
                            errMsg = `Direct calls to initialize() are disabled and used internally by the PipecatClient.`;
                            break;
                        case "initDevices":
                            errMsg = `Direct calls to initDevices() are disabled. Please use the PipecatClient.initDevices() wrapper or let PipecatClient.connect() call it for you.`;
                            break;
                        case "sendReadyMessage":
                            errMsg = `Direct calls to sendReadyMessage() are disabled and used internally by the PipecatClient.`;
                            break;
                        case "connect":
                            errMsg = `Direct calls to connect() are disabled. Please use the PipecatClient.connect() wrapper.`;
                            break;
                        case "disconnect":
                            errMsg = `Direct calls to disconnect() are disabled. Please use the PipecatClient.disconnect() wrapper.`;
                            break;
                    }
                    if (errMsg) return ()=>{
                        throw new Error(errMsg);
                    };
                    // Forward other method calls
                    return (...args)=>{
                        // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
                        return target[prop](...args);
                    };
                }
                // Forward property access
                return Reflect.get(target, prop, receiver);
            }
        });
    }
    get proxy() {
        return this._proxy;
    }
}


var $109c6e1fa5564bb5$exports = {};

$parcel$export($109c6e1fa5564bb5$exports, "learnAboutClient", () => $109c6e1fa5564bb5$export$7eb7b0a641098f31);
$parcel$export($109c6e1fa5564bb5$exports, "messageSizeWithinLimit", () => $109c6e1fa5564bb5$export$48f8227f1e7323f5);


function $109c6e1fa5564bb5$export$7eb7b0a641098f31() {
    let about = {
        library: (0, $8cf63be2a2a16953$exports.name),
        library_version: (0, $8cf63be2a2a16953$exports.version),
        platform_details: {}
    };
    // This uses legacy browser user agent parsing, which we still fall
    // back to if the User Agent Hints API is not available
    let navAgentInfo = null;
    if (window?.navigator?.userAgent) try {
        navAgentInfo = (0, ($parcel$interopDefault($58KWk$bowser))).parse(window.navigator.userAgent);
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    } catch (_) {
    // void
    }
    if (navAgentInfo?.browser?.name) about.platform_details.browser = navAgentInfo.browser.name;
    if (navAgentInfo?.browser?.name === "Safari" && !navAgentInfo.browser.version) about.platform_details.browser_version = "Web View";
    else if (navAgentInfo?.browser?.version) about.platform_details.browser_version = navAgentInfo.browser.version;
    if (navAgentInfo?.platform?.type) about.platform_details.platform_type = navAgentInfo.platform.type;
    if (navAgentInfo?.engine?.name) about.platform_details.engine = navAgentInfo.engine.name;
    if (navAgentInfo?.os) {
        about.platform = navAgentInfo.os.name;
        about.platform_version = navAgentInfo.os.version;
    }
    return about;
}
function $109c6e1fa5564bb5$export$48f8227f1e7323f5(message, maxSize) {
    const getSizeInBytes = (obj)=>{
        const jsonString = JSON.stringify(obj);
        const encoder = new TextEncoder();
        const bytes = encoder.encode(jsonString);
        return bytes.length;
    };
    const size = getSizeInBytes(message);
    return size <= maxSize;
}


var $02f900e1428a591c$var$__decorate = undefined && undefined.__decorate || function(decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for(var i = decorators.length - 1; i >= 0; i--)if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
class $02f900e1428a591c$var$RTVIEventEmitter extends (0, ($parcel$interopDefault($58KWk$events))) {
}
class $02f900e1428a591c$export$8f7f86a77535f7a3 extends $02f900e1428a591c$var$RTVIEventEmitter {
    constructor(options){
        super();
        this._functionCallCallbacks = {};
        this._botTranscriptionWarned = false;
        this._llmFunctionCallWarned = false;
        (0, $b1a2fe36be912934$export$e4036f9b8ddb7379)((0, $109c6e1fa5564bb5$export$7eb7b0a641098f31)());
        this._transport = options.transport;
        this._transportWrapper = new (0, $8e196b651aab6862$export$82b6ede160a64a3c)(this._transport);
        this._disconnectOnBotDisconnect = options.disconnectOnBotDisconnect ?? true;
        // Wrap transport callbacks with event triggers
        // This allows for either functional callbacks or .on / .off event listeners
        const wrappedCallbacks = {
            ...options.callbacks,
            onMessageError: (message)=>{
                options?.callbacks?.onMessageError?.(message);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).MessageError, message);
            },
            onError: (message)=>{
                options?.callbacks?.onError?.(message);
                try {
                    this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).Error, message);
                } catch (e) {
                    if (e instanceof Error && e.message.includes("Unhandled error")) {
                        if (!options?.callbacks?.onError) (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("No onError callback registered to handle error", message);
                    } else (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("Could not emit error", message, e);
                }
                const data = message.data;
                if (data?.fatal) {
                    (0, $42491a06a064f4c0$export$af88d00dbe7f521).error("Fatal error reported. Disconnecting...");
                    this.disconnect();
                }
            },
            onConnected: ()=>{
                options?.callbacks?.onConnected?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).Connected);
            },
            onDisconnected: ()=>{
                options?.callbacks?.onDisconnected?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).Disconnected);
            },
            onTransportStateChanged: (state)=>{
                options?.callbacks?.onTransportStateChanged?.(state);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).TransportStateChanged, state);
            },
            onParticipantJoined: (p)=>{
                options?.callbacks?.onParticipantJoined?.(p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).ParticipantConnected, p);
            },
            onParticipantLeft: (p)=>{
                options?.callbacks?.onParticipantLeft?.(p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).ParticipantLeft, p);
            },
            onTrackStarted: (track, p)=>{
                options?.callbacks?.onTrackStarted?.(track, p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).TrackStarted, track, p);
            },
            onTrackStopped: (track, p)=>{
                options?.callbacks?.onTrackStopped?.(track, p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).TrackStopped, track, p);
            },
            onScreenTrackStarted: (track, p)=>{
                options?.callbacks?.onScreenTrackStarted?.(track, p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).ScreenTrackStarted, track, p);
            },
            onScreenTrackStopped: (track, p)=>{
                options?.callbacks?.onScreenTrackStopped?.(track, p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).ScreenTrackStopped, track, p);
            },
            onScreenShareError: (errorMessage)=>{
                options?.callbacks?.onScreenShareError?.(errorMessage);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).ScreenShareError, errorMessage);
            },
            onAvailableCamsUpdated: (cams)=>{
                options?.callbacks?.onAvailableCamsUpdated?.(cams);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).AvailableCamsUpdated, cams);
            },
            onAvailableMicsUpdated: (mics)=>{
                options?.callbacks?.onAvailableMicsUpdated?.(mics);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).AvailableMicsUpdated, mics);
            },
            onAvailableSpeakersUpdated: (speakers)=>{
                options?.callbacks?.onAvailableSpeakersUpdated?.(speakers);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).AvailableSpeakersUpdated, speakers);
            },
            onCamUpdated: (cam)=>{
                options?.callbacks?.onCamUpdated?.(cam);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).CamUpdated, cam);
            },
            onMicUpdated: (mic)=>{
                options?.callbacks?.onMicUpdated?.(mic);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).MicUpdated, mic);
            },
            onSpeakerUpdated: (speaker)=>{
                options?.callbacks?.onSpeakerUpdated?.(speaker);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).SpeakerUpdated, speaker);
            },
            onDeviceError: (error)=>{
                options?.callbacks?.onDeviceError?.(error);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).DeviceError, error);
            },
            onBotStarted: (botResponse)=>{
                options?.callbacks?.onBotStarted?.(botResponse);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotStarted, botResponse);
            },
            onBotConnected: (p)=>{
                options?.callbacks?.onBotConnected?.(p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotConnected, p);
            },
            onBotReady: (botReadyData)=>{
                options?.callbacks?.onBotReady?.(botReadyData);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotReady, botReadyData);
            },
            onBotDisconnected: (p)=>{
                options?.callbacks?.onBotDisconnected?.(p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotDisconnected, p);
                if (this._disconnectOnBotDisconnect) {
                    (0, $42491a06a064f4c0$export$af88d00dbe7f521).info("Bot disconnected. Disconnecting client...");
                    this.disconnect();
                }
            },
            onUserStartedSpeaking: ()=>{
                options?.callbacks?.onUserStartedSpeaking?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).UserStartedSpeaking);
            },
            onUserStoppedSpeaking: ()=>{
                options?.callbacks?.onUserStoppedSpeaking?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).UserStoppedSpeaking);
            },
            onBotStartedSpeaking: ()=>{
                options?.callbacks?.onBotStartedSpeaking?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotStartedSpeaking);
            },
            onBotStoppedSpeaking: ()=>{
                options?.callbacks?.onBotStoppedSpeaking?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotStoppedSpeaking);
            },
            onRemoteAudioLevel: (level, p)=>{
                options?.callbacks?.onRemoteAudioLevel?.(level, p);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).RemoteAudioLevel, level, p);
            },
            onLocalAudioLevel: (level)=>{
                options?.callbacks?.onLocalAudioLevel?.(level);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).LocalAudioLevel, level);
            },
            onUserMuteStarted: ()=>{
                options?.callbacks?.onUserMuteStarted?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).UserMuteStarted);
            },
            onUserMuteStopped: ()=>{
                options?.callbacks?.onUserMuteStopped?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).UserMuteStopped);
            },
            onUserTranscript: (data)=>{
                options?.callbacks?.onUserTranscript?.(data);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).UserTranscript, data);
            },
            onBotOutput: (data)=>{
                options?.callbacks?.onBotOutput?.(data);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotOutput, data);
            },
            onBotTranscript: (text)=>{
                if (!this._botTranscriptionWarned) {
                    (0, $42491a06a064f4c0$export$af88d00dbe7f521).warn("[Pipecat Client] Bot transcription is deprecated. Please use the onBotOutput instead.");
                    this._botTranscriptionWarned = true;
                }
                options?.callbacks?.onBotTranscript?.(text);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotTranscript, text);
            },
            onBotLlmText: (text)=>{
                options?.callbacks?.onBotLlmText?.(text);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotLlmText, text);
            },
            onBotLlmStarted: ()=>{
                options?.callbacks?.onBotLlmStarted?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotLlmStarted);
            },
            onBotLlmStopped: ()=>{
                options?.callbacks?.onBotLlmStopped?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotLlmStopped);
            },
            onBotTtsText: (text)=>{
                options?.callbacks?.onBotTtsText?.(text);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotTtsText, text);
            },
            onBotTtsStarted: ()=>{
                options?.callbacks?.onBotTtsStarted?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotTtsStarted);
            },
            onBotTtsStopped: ()=>{
                options?.callbacks?.onBotTtsStopped?.();
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotTtsStopped);
            }
        };
        // Update options to reference wrapped callbacks and config defaults
        this._options = {
            ...options,
            callbacks: wrappedCallbacks,
            enableMic: options.enableMic ?? true,
            enableCam: options.enableCam ?? false,
            enableScreenShare: options.enableScreenShare ?? false
        };
        // Instantiate the transport class and bind message handler
        this._initialize();
        // Get package version number
        (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[Pipecat Client] Initialized", this.version);
    }
    setLogLevel(level) {
        (0, $42491a06a064f4c0$export$af88d00dbe7f521).setLevel(level);
    }
    // ------ Transport methods
    /**
     * Initialize local media devices
     */ async initDevices() {
        (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[Pipecat Client] Initializing devices...");
        await this._transport.initDevices();
    }
    /**
     * startBot() is a method that initiates the bot by posting to a specified endpoint
     * that optionally returns connection parameters for establishing a transport session.
     * @param startBotParams
     * @returns Promise that resolves to TransportConnectionParams or unknown
     */ async startBot(startBotParams) {
        if (this._transport.state === "disconnected") await this._transport.initDevices();
        this._transport.state = "authenticating";
        this._transport.startBotParams = startBotParams;
        this._abortController = new AbortController();
        let response;
        try {
            response = await (0, $e93e1075413e9e2e$export$699251e5611cc6db)(startBotParams, this._abortController);
        } catch (e) {
            let errMsg = "An unknown error occurred while starting the bot.";
            let status;
            if (e instanceof Response) {
                const errResp = await e.json();
                errMsg = errResp.info ?? errResp.detail ?? e.statusText;
                status = e.status;
            } else if (e instanceof Error) errMsg = e.message;
            this._options.callbacks?.onError?.(new (0, $b1a2fe36be912934$export$69aa9ab0334b212)((0, $b1a2fe36be912934$export$38b3db05cbf0e240).ERROR_RESPONSE, {
                message: errMsg,
                fatal: true
            }));
            throw new $b4e51e049033e221$export$e7544ab812238a61(errMsg, status);
        }
        this._transport.state = "authenticated";
        this._options.callbacks?.onBotStarted?.(response);
        return response;
    }
    /**
     * The `connect` function establishes a transport session and awaits a
     * bot-ready signal, handling various connection states and errors.
     * @param {TransportConnectionParams} [connectParams] -
     * The `connectParams` parameter in the `connect` method should be of type
     * `TransportConnectionParams`. This parameter is passed to the transport
     * for establishing a transport session.
     * NOTE: `connectParams` as type `ConnectionEndpoint` IS NOW DEPRECATED. If you
     * want to authenticate and connect to a bot in one step, use
     * `startBotAndConnect()` instead.
     * @returns The `connect` method returns a Promise that resolves to an unknown value.
     */ async connect(connectParams) {
        if (connectParams && (0, $e93e1075413e9e2e$export$2dd7ca293b2783)(connectParams)) {
            (0, $42491a06a064f4c0$export$af88d00dbe7f521).warn("Calling connect with an API endpoint is deprecated. Use startBotAndConnect() instead.");
            return this.startBotAndConnect(connectParams);
        }
        // Establish transport session and await bot ready signal
        return new Promise((resolve, reject)=>{
            (async ()=>{
                this._connectResolve = resolve;
                if (this._transport.state === "disconnected") await this._transport.initDevices();
                try {
                    await this._transport.connect(connectParams);
                    await this._transport.sendReadyMessage();
                } catch (e) {
                    this.disconnect();
                    reject(e);
                    return;
                }
            })();
        });
    }
    async startBotAndConnect(startBotParams) {
        const connectionParams = await this.startBot(startBotParams);
        return this.connect(connectionParams);
    }
    /**
     * Disconnect the voice client from the transport
     * Reset / reinitialize transport and abort any pending requests
     */ async disconnect() {
        await this._transport.disconnect();
        this._messageDispatcher.disconnect();
    }
    /**
     * The _initialize function performs internal set up of the transport and
     * message dispatcher.
     */ _initialize() {
        this._transport.initialize(this._options, this.handleMessage.bind(this));
        // Create a new message dispatch queue for async message handling
        this._messageDispatcher = new (0, $3b981481778defd6$export$e9a960646cc432aa)(this._sendMessage.bind(this));
    }
    /**
     * Internal wrapper around the transport's sendMessage method
     */ _sendMessage(message) {
        if (!(0, $109c6e1fa5564bb5$export$48f8227f1e7323f5)(message, this._transport.maxMessageSize)) {
            const msg = `Message data too large. Max size is ${this._transport.maxMessageSize}`;
            this._options.callbacks?.onError?.((0, $b1a2fe36be912934$export$69aa9ab0334b212).error(msg, false));
            throw new $b4e51e049033e221$export$78e1011ee1942cf6(msg);
        }
        try {
            this._transport.sendMessage(message);
        } catch (error) {
            if (error instanceof Error) this._options.callbacks?.onError?.((0, $b1a2fe36be912934$export$69aa9ab0334b212).error(error.message, false));
            else this._options.callbacks?.onError?.((0, $b1a2fe36be912934$export$69aa9ab0334b212).error("Unknown error sending message", false));
            throw error;
        }
    }
    /**
     * Get the current state of the transport
     */ get connected() {
        return [
            "connected",
            "ready"
        ].includes(this._transport.state);
    }
    get transport() {
        return this._transportWrapper.proxy;
    }
    get state() {
        return this._transport.state;
    }
    get version() {
        return (0, (/*@__PURE__*/$parcel$interopDefault($8cf63be2a2a16953$exports))).version;
    }
    // ------ Device methods
    async getAllMics() {
        return await this._transport.getAllMics();
    }
    async getAllCams() {
        return await this._transport.getAllCams();
    }
    async getAllSpeakers() {
        return await this._transport.getAllSpeakers();
    }
    get selectedMic() {
        return this._transport.selectedMic;
    }
    get selectedCam() {
        return this._transport.selectedCam;
    }
    get selectedSpeaker() {
        return this._transport.selectedSpeaker;
    }
    updateMic(micId) {
        this._transport.updateMic(micId);
    }
    updateCam(camId) {
        this._transport.updateCam(camId);
    }
    updateSpeaker(speakerId) {
        this._transport.updateSpeaker(speakerId);
    }
    enableMic(enable) {
        this._transport.enableMic(enable);
    }
    get isMicEnabled() {
        return this._transport.isMicEnabled;
    }
    enableCam(enable) {
        this._transport.enableCam(enable);
    }
    get isCamEnabled() {
        return this._transport.isCamEnabled;
    }
    tracks() {
        return this._transport.tracks();
    }
    enableScreenShare(enable) {
        return this._transport.enableScreenShare(enable);
    }
    get isSharingScreen() {
        return this._transport.isSharingScreen;
    }
    // ------ Messages
    /**
     * Directly send a message to the bot via the transport.
     * Do not await a response.
     * @param msgType - a string representing the message type
     * @param data - a dictionary of data to send with the message
     */ sendClientMessage(msgType, data) {
        this._sendMessage(new (0, $b1a2fe36be912934$export$69aa9ab0334b212)((0, $b1a2fe36be912934$export$38b3db05cbf0e240).CLIENT_MESSAGE, {
            t: msgType,
            d: data
        }));
    }
    /**
     * Directly send a message to the bot via the transport.
     * Wait for and return the response.
     * @param msgType - a string representing the message type
     * @param data - a dictionary of data to send with the message
     * @param timeout - optional timeout in milliseconds for the response
     */ async sendClientRequest(msgType, data, timeout) {
        const msgData = {
            t: msgType,
            d: data
        };
        const response = await this._messageDispatcher.dispatch(msgData, (0, $b1a2fe36be912934$export$38b3db05cbf0e240).CLIENT_MESSAGE, timeout);
        const ret_data = response.data;
        return ret_data.d;
    }
    registerFunctionCallHandler(functionName, callback) {
        this._functionCallCallbacks[functionName] = callback;
    }
    unregisterFunctionCallHandler(functionName) {
        delete this._functionCallCallbacks[functionName];
    }
    unregisterAllFunctionCallHandlers() {
        this._functionCallCallbacks = {};
    }
    async appendToContext(context) {
        (0, $42491a06a064f4c0$export$af88d00dbe7f521).warn("appendToContext() is deprecated. Use sendText() instead.");
        await this._sendMessage(new (0, $b1a2fe36be912934$export$69aa9ab0334b212)((0, $b1a2fe36be912934$export$38b3db05cbf0e240).APPEND_TO_CONTEXT, {
            role: context.role,
            content: context.content,
            run_immediately: context.run_immediately
        }));
        return true;
    }
    async sendText(content, options = {}) {
        await this._sendMessage(new (0, $b1a2fe36be912934$export$69aa9ab0334b212)((0, $b1a2fe36be912934$export$38b3db05cbf0e240).SEND_TEXT, {
            content: content,
            options: options
        }));
    }
    /**
     * Disconnects the bot, but keeps the session alive
     */ disconnectBot() {
        this._sendMessage(new (0, $b1a2fe36be912934$export$69aa9ab0334b212)((0, $b1a2fe36be912934$export$38b3db05cbf0e240).DISCONNECT_BOT, {}));
    }
    handleMessage(ev) {
        (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[RTVI Message]", ev);
        switch(ev.type){
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_READY:
                {
                    const data = ev.data;
                    const botVersion = data.version ? data.version.split(".").map(Number) : [
                        0,
                        0,
                        0
                    ];
                    (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug(`[Pipecat Client] Bot is ready. Version: ${data.version}`);
                    if (botVersion[0] < 1) (0, $42491a06a064f4c0$export$af88d00dbe7f521).warn("[Pipecat Client] Bot version is less than 1.0.0, which may not be compatible with this client.");
                    this._connectResolve?.(ev.data);
                    this._options.callbacks?.onBotReady?.(ev.data);
                    break;
                }
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).ERROR:
                this._options.callbacks?.onError?.(ev);
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).SERVER_RESPONSE:
                this._messageDispatcher.resolve(ev);
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).ERROR_RESPONSE:
                {
                    const resp = this._messageDispatcher.reject(ev);
                    this._options.callbacks?.onMessageError?.(resp);
                    break;
                }
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).USER_STARTED_SPEAKING:
                this._options.callbacks?.onUserStartedSpeaking?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).USER_STOPPED_SPEAKING:
                this._options.callbacks?.onUserStoppedSpeaking?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_STARTED_SPEAKING:
                this._options.callbacks?.onBotStartedSpeaking?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_STOPPED_SPEAKING:
                this._options.callbacks?.onBotStoppedSpeaking?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).USER_MUTE_STARTED:
                this._options.callbacks?.onUserMuteStarted?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).USER_MUTE_STOPPED:
                this._options.callbacks?.onUserMuteStopped?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).USER_TRANSCRIPTION:
                {
                    const TranscriptData = ev.data;
                    this._options.callbacks?.onUserTranscript?.(TranscriptData);
                    break;
                }
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_OUTPUT:
                this._options.callbacks?.onBotOutput?.(ev.data);
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_TRANSCRIPTION:
                this._options.callbacks?.onBotTranscript?.(ev.data);
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_LLM_TEXT:
                this._options.callbacks?.onBotLlmText?.(ev.data);
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_LLM_STARTED:
                this._options.callbacks?.onBotLlmStarted?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_LLM_STOPPED:
                this._options.callbacks?.onBotLlmStopped?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_TTS_TEXT:
                this._options.callbacks?.onBotTtsText?.(ev.data);
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_TTS_STARTED:
                this._options.callbacks?.onBotTtsStarted?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_TTS_STOPPED:
                this._options.callbacks?.onBotTtsStopped?.();
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).METRICS:
                this._options.callbacks?.onMetrics?.(ev.data);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).Metrics, ev.data);
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).SERVER_MESSAGE:
                this._options.callbacks?.onServerMessage?.(ev.data);
                this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).ServerMessage, ev.data);
                break;
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).LLM_FUNCTION_CALL_STARTED:
                {
                    const data = ev.data;
                    this._options.callbacks?.onLLMFunctionCallStarted?.(data);
                    this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).LLMFunctionCallStarted, data);
                    break;
                }
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).LLM_FUNCTION_CALL_IN_PROGRESS:
                {
                    const data = ev.data;
                    this._maybeTriggerFunctionCallCallback(data);
                    this._options.callbacks?.onLLMFunctionCallInProgress?.(data);
                    this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).LLMFunctionCallInProgress, data);
                    break;
                }
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).LLM_FUNCTION_CALL_STOPPED:
                {
                    const data = ev.data;
                    this._options.callbacks?.onLLMFunctionCallStopped?.(data);
                    this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).LLMFunctionCallStopped, data);
                    break;
                }
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).LLM_FUNCTION_CALL:
                {
                    const data = ev.data;
                    const inProgressData = {
                        function_name: data.function_name,
                        tool_call_id: data.tool_call_id,
                        arguments: data.args
                    };
                    this._maybeTriggerFunctionCallCallback(inProgressData);
                    if (this._options.callbacks?.onLLMFunctionCall) {
                        if (!this._llmFunctionCallWarned) {
                            (0, $42491a06a064f4c0$export$af88d00dbe7f521).warn("[Pipecat Client] onLLMFunctionCall is deprecated. Please use onLLMFunctionCallInProgress instead.");
                            this._llmFunctionCallWarned = true;
                        }
                    }
                    this._options.callbacks?.onLLMFunctionCall?.(data);
                    this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).LLMFunctionCall, data);
                    break;
                }
            case (0, $b1a2fe36be912934$export$38b3db05cbf0e240).BOT_LLM_SEARCH_RESPONSE:
                {
                    const data = ev.data;
                    this._options.callbacks?.onBotLlmSearchResponse?.(data);
                    this.emit((0, $82c0e475b1b90003$export$6b4624d233c61fcb).BotLlmSearchResponse, data);
                    break;
                }
            default:
                (0, $42491a06a064f4c0$export$af88d00dbe7f521).debug("[Pipecat Client] Unrecognized message type", ev.type);
                break;
        }
    }
    _maybeTriggerFunctionCallCallback(data) {
        // Function call callbacks are meant only for function calls that need information
        // from the client to complete and generate a result. This process requires that
        // the event includes the function name. For client-side logic meant simply to
        // react to the fact that a function call is happening, you should use the
        // traditional onLLMFunctionCallStarted/InProgress/Stopped events instead.
        if (!data.function_name) return;
        const fc = this._functionCallCallbacks[data.function_name];
        if (fc) {
            const params = {
                functionName: data.function_name ?? "",
                arguments: data.arguments ?? {}
            };
            fc(params).then((result)=>{
                if (result == undefined) return;
                this._sendMessage(new (0, $b1a2fe36be912934$export$69aa9ab0334b212)((0, $b1a2fe36be912934$export$38b3db05cbf0e240).LLM_FUNCTION_CALL_RESULT, {
                    function_name: data.function_name,
                    tool_call_id: data.tool_call_id,
                    arguments: data.arguments ?? {},
                    result: result
                }));
            }).catch((error)=>{
                (0, $42491a06a064f4c0$export$af88d00dbe7f521).error("Error in function call callback", error);
            });
        }
    }
}
$02f900e1428a591c$var$__decorate([
    (0, $37a63b97d182bc86$export$ebc0d747cf8770bc)
], $02f900e1428a591c$export$8f7f86a77535f7a3.prototype, "startBot", null);
$02f900e1428a591c$var$__decorate([
    (0, $37a63b97d182bc86$export$ebc0d747cf8770bc)
], $02f900e1428a591c$export$8f7f86a77535f7a3.prototype, "connect", null);
$02f900e1428a591c$var$__decorate([
    (0, $37a63b97d182bc86$export$ebc0d747cf8770bc)
], $02f900e1428a591c$export$8f7f86a77535f7a3.prototype, "startBotAndConnect", null);
$02f900e1428a591c$var$__decorate([
    (0, $37a63b97d182bc86$export$f1586721024c4dab)
], $02f900e1428a591c$export$8f7f86a77535f7a3.prototype, "sendClientMessage", null);
$02f900e1428a591c$var$__decorate([
    (0, $37a63b97d182bc86$export$f1586721024c4dab)
], $02f900e1428a591c$export$8f7f86a77535f7a3.prototype, "sendClientRequest", null);
$02f900e1428a591c$var$__decorate([
    (0, $37a63b97d182bc86$export$f1586721024c4dab)
], $02f900e1428a591c$export$8f7f86a77535f7a3.prototype, "appendToContext", null);
$02f900e1428a591c$var$__decorate([
    (0, $37a63b97d182bc86$export$f1586721024c4dab)
], $02f900e1428a591c$export$8f7f86a77535f7a3.prototype, "sendText", null);
$02f900e1428a591c$var$__decorate([
    (0, $37a63b97d182bc86$export$f1586721024c4dab)
], $02f900e1428a591c$export$8f7f86a77535f7a3.prototype, "disconnectBot", null);







$parcel$exportWildcard($c2758f3146a53d04$exports, $02f900e1428a591c$exports);
$parcel$exportWildcard($c2758f3146a53d04$exports, $3b981481778defd6$exports);
$parcel$exportWildcard($c2758f3146a53d04$exports, $42491a06a064f4c0$exports);
$parcel$exportWildcard($c2758f3146a53d04$exports, $e93e1075413e9e2e$exports);
$parcel$exportWildcard($c2758f3146a53d04$exports, $8e196b651aab6862$exports);
$parcel$exportWildcard($c2758f3146a53d04$exports, $109c6e1fa5564bb5$exports);



$parcel$exportWildcard(module.exports, $c2758f3146a53d04$exports);
$parcel$exportWildcard(module.exports, $609db2a766ef81cc$exports);
